File: hackerrank/algorithms/plusMinus.py
# https://www.hackerrank.com/challenges/plus-minus/

def plusMinus(arr):
    arr.sort()

    negative_count = 0
    zero_count = 0
    positive_count = 0
    length = len(arr)

    for _, item in enumerate(arr):
        if item < 0:
            negative_count += 1
        elif item == 0:
            zero_count += 1
        elif item > 0:
            positive_count += 1
    
    print(positive_count / length)
    print(negative_count / length)
    print(zero_count / length)

arr = [-4, 3, -9, 0, 4, 1]
plusMinus(arr)

File: hackerrank/plusMinus.java
// https://www.hackerrank.com/challenges/plus-minus/problem
import java.util.Collections;
import java.util.List;

class Result {

    public static void plusMinus(List<Integer> arr) {
        Collections.sort(arr);

        Float positive = 0f;
        Float negative = 0f;
        Float zero = 0f;

        for (int i = 0; i < arr.size(); i++) {
            if (arr.get(i) > 0) {
                positive++;
            } else if (arr.get(i) < 0) {
                negative++;
            } else if (arr.get(i) == 0) {
                zero++;
            }
        }

        System.out.println(positive / arr.size());
        System.out.println(negative / arr.size());
        System.out.println(zero / arr.size());
    }

}